### PR TITLE
fix: [PL-39331]: update email to lowerCase when signup with OAuth

### PR DESCRIPTION
**Pull request by pbarapatre10**
## Describe your changes
In OAuth signup flow currently, email is stored as is, without converting to lowerCase. Saving email in lower case is the norm in user management. With email in users not case insensitive unique, this led to issue having same email (different cases) been stored in Harness users collection. We want to make the email in users collection as case insensitive unique, but that involves updating existing index in users collection, where in we are having some challenges (dropping existing indexes through db query and re-creating new one, but this may lead to issue in that small duration when index doesn't exist at all and traffic might still flow in). This is a first step fix towards ensuring email are stored in lowercase during OAuth sign up as well.

Testing done:
- Tested locally

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53132)

### CODE REVIEW

**Comment by pbarapatre10** [comment]
trigger uts

File: 920-ng-signup/src/main/java/io/harness/signup/services/impl/SignupServiceImpl.java [hunk 1]
```diff
@@ -498,6 +498,7 @@ public UserInfo oAuthSignup(OAuthSignupDTO dto) {
       throw e;
     }
 
+    dto.setEmail(dto.getEmail().toLowerCase());
```
Comment (kapil-harness) [inline 920-ng-signup/src/main/java/io/harness/signup/services/impl/SignupServiceImpl.java:501]: just curious, why not create a custom getter instead? Which would return email in lower case.

Comment (pbarapatre10) [reply 920-ng-signup/src/main/java/io/harness/signup/services/impl/SignupServiceImpl.java:501]: @kapil-harness if you check 5-6 lines above (line 495 `validateEmail()`, where we are validating the supplied email in request, which also uses the input unmodified value. If we override the getter here, we would get the changed (all lower-cased) email as input to the email validator as well. I'm not sure what all validations happen inside there and any domain check are case-sensitive/insensitive, etc, but on overview they look case sensitive checks. If we are sure about passing lower-case email won't impact in any way the validateEmail logic, we can go ahead with getter to return lowerCase always. Thanks

**Review by kapil-harness — APPROVED** [review]

**Review by shashank-harness — APPROVED** [review]
Lgtm

**Comment by pbarapatre10** [comment]
trigger messagecheck

**Pull request merged**